test/semantics/assign03.f90

```diff
@@ -143,18 +143,25 @@ subroutine s10
     type(t), target :: x
     type(t), target :: y(10,10)
     p(1:16) => x%a
+    p(1:8) => x%a(:,3:4)
     p(1:1) => x%b  ! We treat scalars as simply contiguous
+    p(1:1) => x%a(1,1)
+    p(1:1) => y(1,1)%a(1,1)
+    p(1:1) => y(:,1)%a(1,1)  ! Rank 1 RHS
```

Comment:
This case is not simply contiguous.  See 9.5.4; the final *part-ref* here isn't the one with rank > 0.

As I understand this part of the standard, there really can't be anything to the right of the *part-ref* with rank in a simply contiguous *data-ref* (no components, subscripted or otherwise; no complex components; no substrings).  And other constraints prevent multiple *part-ref*s from having rank > 0.